Mathlib/FieldTheory/RatFunc/AsPolynomial.lean

```diff
@@ -193,3 +198,47 @@ theorem eval_mul {x y : RatFunc K} (hx : Polynomial.eval₂ f a (denom x) ≠ 0)
 end Field
 
 end Eval
+
+end RatFunc
+
+section AdicValuation
+
+variable (K : Type _) [Field K]
+
+namespace Polynomial
+
+open IsDedekindDomain.HeightOneSpectrum
+
+/-- This is the principal ideal generated by `X` in the ring of polynomials over a field K,
+  regarded as an element of the height-one-spectrum. -/
+def idealX : IsDedekindDomain.HeightOneSpectrum K[X] where
+  asIdeal := Ideal.span {Polynomial.X}
```

Comment:
```suggestion
  asIdeal := Ideal.span {X}
```
Untested, but if it works you can do the same everywhere else.

Reply:
It works